src/directives/badge.js

```diff
@@ -9,10 +9,18 @@ function directive (el, binding) {
       icon: false,
       left: false,
       bottom: false,
-      overlap: false
+      overlap: false,
+      visible: true
     }
   )
 
+  if (!config.visible || (binding.expression && !binding.value)) {
```

Comment:
I like the way you did this, instead of creating a new class that hides it, you remove all applicable classes.

Question, what do you think about just adding one of the display classes instead? Would be more future proof as the names badge--left, badge--icon could change or be added to in the future.